README.md

```diff
@@ -460,6 +460,28 @@ rescue JWT::InvalidSubError
 end
 ```
 
+### Finding a Key
+
+To dynamically find the secret that was used for signing the JWT, pass a block to the decode block. The block receives headers and the original payload as parameters. It should return with the secret to verify the signature that was used to sign the JWT.
+
+```ruby
+iss = 'My_Awesome_Company'
+iss_payload = { data: 'data', iss: iss }
+
+secrets = { iss => hmac_secret }
+
+token = JWT.encode iss_payload, hmac_secret, 'HS256'
+
+begin
+  # Add iss to the validation to check if the token has been manipulated
+  decoded_token = JWT.decode(token, nil, true, { iss: iss, verify_iss: true, algorithm: 'HS256' }) do |_headers, payload|
```

Comment:
Just a random idea: Would i make sense to have 2 different issuers to pick between in the example.